app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -176,6 +186,22 @@ class PersonalDetailsCollectionController @Inject()(personalDetailsSubmission: P
     }
   }
 
+  private def sessionKeyCheck(result: Future[Result], completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl])(implicit request: MessagesRequest[AnyContent]): Future[Result] = {
```

Comment:
Adding to an already complex controller will make it harder to seperate these methods into seperate files further down the line. BUT we are following the pattern of every other controller method by adding a private method here